mne/decoding/tests/test_ssd.py

```diff
@@ -0,0 +1,240 @@
+                                  l_trans_bandwidth=1, h_trans_bandwidth=1,
+                                  fir_design='firwin')
+    filt_params_noise = dict(l_freq=freqs_noise[0], h_freq=freqs_noise[1],
+                                 l_trans_bandwidth=1, h_trans_bandwidth=1,
+                                 fir_design='firwin')
+    ssd = SSD(info, filt_params_signal, filt_params_noise)
+    raw = io.RawArray(X, info)
+
+    pytest.raises(ValueError, ssd.fit, raw)    
+    
+    # Fit
+    n_components=10
+    filt_params_signal=dict(l_freq=freqs_sig[0], h_freq=freqs_sig[1],
+                                  l_trans_bandwidth=1, h_trans_bandwidth=1,
+                                  fir_design='firwin')
+    filt_params_noise=dict(l_freq=freqs_noise[0], h_freq=freqs_noise[1],
+                                 l_trans_bandwidth=1, h_trans_bandwidth=1,
+                                 fir_design='firwin')
+    ssd = SSD(info, filt_params_signal, filt_params_noise,
+              n_components=n_components)
+    ssd.fit(X)
+
+    assert (ssd.filters_.shape == (n_channels, n_channels))
+    assert (ssd.patterns_.shape == (n_channels, n_channels))
+
+    # Transform
+    X_ssd = ssd.fit_transform(X)
+    assert (X_ssd.shape[0] == n_components)
+    # back and forward
+    ssd = SSD(info, filt_params_signal, filt_params_noise,
+              n_components=None, sort_by_spectral_ratio=False)
+    ssd.fit(X)
+    X_denoised = ssd.inverse_transform(X)
+    assert_array_almost_equal(X_denoised, X)
+    
+    # Power ratio ordering
+    spec_ratio, sorter_spec=get_spectral_ratio(ssd.transform(X), sf, ssd)
+    # since we now that the number of true components is 5, the relative
+    # difference should be low for the first 5 and then increases
+    index_diff=np.argmax(-np.diff(spec_ratio))
+    assert index_diff == n_components_true-1
+
+    # Check detected peaks
+    # fit ssd
+    n_components = n_components_true # we know that are 5
+    filt_params_signal = dict(l_freq=freqs_sig[0], h_freq=freqs_sig[1],
+                                  l_trans_bandwidth=1, h_trans_bandwidth=1,
+                                  fir_design='firwin')
+    filt_params_noise = dict(l_freq=freqs_noise[0], h_freq=freqs_noise[1],
+                                 l_trans_bandwidth=1, h_trans_bandwidth=1,
+                                 fir_design='firwin')
+    ssd = SSD(info, filt_params_signal, filt_params_noise,
+              n_components=n_components, sort_by_spectral_ratio=False)
+    ssd.fit(X)
+   
+    out = ssd.transform(X)
+    psd_out, _ = psd_array_welch(out[0], sfreq=250, n_fft=250)
+    psd_S, _ = psd_array_welch(S[0], sfreq=250, n_fft=250)
+    corr = np.abs(np.corrcoef(psd_out, psd_S)[0, 1])
+    assert np.abs(corr) > 0.95
+    
+    # Check pattern estimation   
+    # Since there is no exact ordering of the recovered patterns 
+    # a pair-wise greedy search will be done
+    err=[]
+        
+    for i in range(n_channels):
```

Comment:
better: ii than i; in the debugger many special commends have single character names